From: Chad Dupuis <cdupuis@redhat.com>
Date: Tue, 30 Nov 2010 19:59:57 -0500
Subject: [scsi] qla4xxx: msi init request_irq parameter usage fix
Message-id: <20101130195957.10450.46839.sendpatchset@localhost.localdomain>
Patchwork-id: 29703
O-Subject: [RHEL 5.6 PATCH 12/28] qla4xxx: Change MSI initialization from using
	incorrect request_irq parameter.
Bugzilla: 656999

Bugzilla
--------

Bug 656999 (https://bugzilla.redhat.com/show_bug.cgi?id=656999)

Upstream Status
---------------

To be sent upstream.

Description
-----------

>From 051d30c68e091318e2bb56b14e6a0d422d02862e Mon Sep 17 00:00:00 2001
From: Shyam Sundar <shyam.sundar@qlogic.com>
Date: Sat, 20 Nov 2010 14:36:31 -0800
Subject: [PATCH 13/29] qla4xxx: Change MSI initialization from using incorrect request_irq parameter.

IRQF_SHARED flag should not be set when calling request_irq for MSI since
this interrupt mechanism cannot be shared like standard INTx

Signed-off-by: Shyam Sundar <shyam.sundar@qlogic.com>
Signed-off-by: Prasanna Mumbai <prasanna.mumbai@qlogic.com>

diff --git a/drivers/scsi/qla4xxx/ql4_isr.c b/drivers/scsi/qla4xxx/ql4_isr.c
index 284c92d..6b2111e 100644
--- a/drivers/scsi/qla4xxx/ql4_isr.c
+++ b/drivers/scsi/qla4xxx/ql4_isr.c
@@ -1148,7 +1148,7 @@ try_msi:
 	ret = pci_enable_msi(ha->pdev);
 	if (!ret) {
 		ret = request_irq(ha->pdev->irq, qla4_8xxx_msi_handler,
-		    IRQF_DISABLED|IRQF_SHARED, DRIVER_NAME, ha);
+		    IRQF_DISABLED, DRIVER_NAME, ha);
 		if (!ret) {
 			DEBUG2(printk(KERN_INFO"MSI: Enabled.\n"));
 			set_bit(AF_MSI_ENABLED, &ha->flags);
